deps/graph-parser/src/logseq/graph_parser/text.cljs

```diff
@@ -213,49 +92,95 @@
     (re-find #"^\d+$" v)
     (parse-long v)))
 
-(def ^:private page-ref-or-tag-re
-  (re-pattern (str "#?" (page-ref/->page-ref-re-str "(.*?)") "|"
-                   ;; Don't capture punctuation at end of a tag
-                   "#([\\S]+[^\\s.!,])")))
+(defn get-ref-from-ast
+  [[typ data]]
+  (case typ
+    "Link"
+    (case (first (:url data))
+      "Page_ref"
+      (second (:url data))
 
-(defn extract-page-refs-and-tags
-  "Returns set of page-refs and tags in given string or returns string if none
-  are found"
-  [string]
-  (let [refs (map #(or (second %) (get % 2))
-                  (re-seq page-ref-or-tag-re string))]
-    (if (seq refs) (set refs) string)))
+      "Search"
+      (second (:url data))
 
-(defn parse-property
-  "Property value parsing that takes into account built-in properties, format
-  and user config"
-  ([k v config-state]
-   (parse-property :markdown k v config-state))
-  ([format k v config-state]
-   (let [k (name k)
-         v (if (or (symbol? v) (keyword? v)) (name v) (str v))
-         v (string/trim v)]
-     (cond
-       (contains? (set/union
-                   #{"title" "filters"}
-                   (get config-state :ignored-page-references-keywords)) k)
-       v
+      nil)
 
-       (gp-util/wrapped-by-quotes? v) ; wrapped in ""
-       v
+    "Nested_link"
+    (page-ref/get-page-name (:content data))
 
-       (contains? @non-parsing-properties (string/lower-case k))
-       v
+    "Tag"
+    (if (= "Plain" (ffirst data))
+      (second (first data))
+      (get-ref-from-ast (first data)))
 
-       (gp-mldoc/link? format v)
-       v
+    nil))
 
-       (contains? gp-property/editable-linkable-built-in-properties (keyword k))
-       (split-page-refs-without-brackets v)
+(defn extract-refs-from-mldoc-ast
+  [v]
+  (->> v
+       (remove gp-mldoc/ast-link?)
+       (keep get-ref-from-ast)
+       (map (fn [x]
+              (string/trim x)))
+       (set)))
 
-       :else
-       (if-some [res (parse-non-string-property-value v)]
-         res
-         (if (:rich-property-values? config-state)
-           (extract-page-refs-and-tags v)
-           (split-page-refs-without-brackets v)))))))
+(defn sep-by-comma
+  [s]
+  (when s
+    (some->>
+     (string/split s #"[\,|，]{1}")
+     (remove string/blank?)
+     (map string/trim))))
+
+(defn separated-by-commas?
+  [config-state k]
+  (let [k' (if (keyword? k) k (keyword k))]
+    (contains? (set/union #{:alias :tags}
+                          (set (get config-state :property/separated-by-commas))
+                          (set (:property/separated-by-commas config-state)))
+               k')))
+
+(defn parse-property
+  "Property value parsing that takes into account built-in properties, and user config"
+  [k v mldoc-ast config-state]
+  (let [refs (extract-refs-from-mldoc-ast mldoc-ast)
+        property-separated-by-commas? (and (separated-by-commas? config-state k)
+                                           (empty? refs))
```

Comment:
Thanks for pointing this out! 
Initially, I don't want to support these mixed values because I'm afraid that the old regex code is still needed 😂 , but later on, I realized that mldoc ast can be used to support this need, the change is included in 50db6e42a.

Reply:
Nicely done! I found one behavior that probably isn't intended but it shouldn't matter as it doesn't prevent any comma separated use case from working correctly e.g. `tags:: foo [[bar]] ok -> #{"foo" "bar" "ok"}`